odoo/addons/test_testing_utilities/tests/test_freeze_time.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import time
+
+from datetime import datetime, date, timedelta
+from pytz import timezone
+
+from odoo.tools import freeze_time
+from odoo.tests.common import TransactionCase
+from odoo import fields
+
+
+class FreezeTimeTests(TransactionCase):
+
+    def test_freeze_datetime(self):
+        with freeze_time('2018-12-03 09:42:35'):
+            frozen_dt = datetime.now()
+            frozen_d = datetime.today()
+            frozen_utc = datetime.utcnow()
+
+        manual_dt = datetime(2018, 12, 3, 9, 42, 35)
+        self.assertEqual(manual_dt, frozen_dt)
+        self.assertEqual(manual_dt, frozen_d)
+        self.assertEqual(manual_dt, frozen_utc)
+
+    def test_freeze_date(self):
+        with freeze_time('2018-12-03'):
+            frozen = date.today()
+
+        manual = date(2018, 12, 3)
+        self.assertEqual(frozen, manual)
+
+    def test_freeze_time(self):
+        with freeze_time('2018-12-03'):
+            frozen_t = time.time()
+            frozen_strft = time.strftime("%a, %d-%b-%Y")
+
+        manual_t = datetime(2018, 12, 3).timestamp()
+        manual_strft = datetime(2018, 12, 3).strftime("%a, %d-%b-%Y")
+        self.assertEqual(frozen_t, manual_t)
+        self.assertEqual(frozen_strft, manual_strft)
+
+    def test_context_mgr_freeze(self):
+        real_time = fields.Datetime.now()
+
+        with freeze_time('2018-12-03 09:05:25'):
+            fake_time = fields.Datetime.now()
+
+        self.assertNotEqual(real_time, fake_time)
+        self.assertEqual(fields.Datetime.to_string(fake_time), '2018-12-03 09:05:25')
+
+    @freeze_time('2018-12-03')
+    def test_func_decorator_freeze(self):
+        self.assertEqual(fields.Date.to_string(fields.Date.today()), '2018-12-03')
+
+    def test_manual_freeze(self):
+        patcher = freeze_time('2018-12-03 09:15:45')
+        real_time = fields.Datetime.now()
+
+        patcher.apply()
+        fake_time = fields.Datetime.now()
+        patcher.revert()
+
+        self.assertNotEqual(real_time, fake_time)
+        self.assertEqual(fields.Datetime.to_string(fake_time), '2018-12-03 09:15:45')
+
+    def test_class_decorator_good_freeze(self):
+        @freeze_time('2018-12-02')
+        class Foo(TransactionCase):
+            def test_time(self):
+                self.assertEqual(datetime.now, datetime(2018, 12, 2))
+
+        Foo()
```

Comment:
Wait, is this actually executed?

Reply:
I guess it is not the point...
You are only checking that one can decorate the class, right?

Reply:
Indeed!

Reply:
Try to call `unittest` to run this class, then!